src/client_side.cc

```diff
@@ -594,7 +594,19 @@ ConnStateData::swanSong()
     flags.readMore = false;
     clientdbEstablished(clientConnection->remote, -1);  /* decrement */
 
-    terminateAll(ERR_NONE, LogTagsErrors());
+    const char *errDetail = stoppedReceiving();
+    auto errType = errDetail ? ERR_REQUEST_STOPPED_RECEIVING : ERR_NONE;
+    if (!errType) {
+        errDetail = stoppedSending();
+        errType = errDetail ? ERR_REQUEST_STOPPED_SENDING : ERR_NONE;
+    }
+    Error error(errType);
+    if (errDetail) {
+        const auto d = MakeNamedErrorDetail(errDetail);
```

Comment:
MakeNamedErrorDetail() is meant for creating reusable details with constant, known to the caller names. Note how all existing MakeNamedErrorDetail() callers store the allocated pointer in a static variable for reuse...

We should not use MakeNamedErrorDetail() API for strings returned by stoppedReceiving() and stoppedSending() -- they are not error detail names and they are not known to the MakeNamedErrorDetail() caller.

Overall, it feels like this code still tries to guess what the earlier error/problem was. Yes, we moved that guess from logging code to swanSong(), but we are still guessing what has happened earlier. In other words, we are still guessing why swanSong() got called. Assuming these guesses correspond to some cases where official Squid code is currently missing known error details, can we detail those cases _earlier_, where/when those details actually become known?